java-lib/src/main/java/org/flinkextended/clink/feature/onehotencoder/ClinkOneHotEncoderModel.java

```diff
@@ -94,67 +112,23 @@ public ClinkOneHotEncoderModel() {
-        DataStream<Tuple2<Integer, Integer>> modelStream =
-                OneHotEncoderModelData.getModelDataStream(modelDataTable);
-
-        GenerateOutputsFunction mapFunction =
-                new GenerateOutputsFunction(getParamMap(), broadcastModelKey, inputCols);
-
-        Function<List<DataStream<?>>, DataStream<Row>> function =
-                dataStreams -> {
-                    DataStream stream = dataStreams.get(0);
-                    return stream.map(mapFunction, outputTypeInfo);
-                };
-
-        DataStream<Row> output =
-                BroadcastUtils.withBroadcastStream(
-                        Collections.singletonList(input),
-                        Collections.singletonMap(broadcastModelKey, modelStream),
-                        function);
-
+        DataStream<Row> output = input.map(new GenerateOutputsFunction(), outputTypeInfo);
         Table outputTable = tEnv.fromDataStream(output);
-
         return new Table[] {outputTable};
     }
 
-    @Override
-    public ClinkOneHotEncoderModel setModelData(Table... inputs) {
-        Preconditions.checkArgument(inputs.length == 1);
-        modelDataTable = inputs[0];
-        return this;
-    }
+    private class GenerateOutputsFunction extends RichMapFunction<Row, Row> {
+        private final String[] inputCols = getInputCols();
+        private Pointer modelPointer;
 
-    @Override
-    public Table[] getModelData() {
-        return new Table[] {modelDataTable};
-    }
-
-    @Override
-    public Map<Param<?>, Object> getParamMap() {
-        return paramMap;
-    }
-
-    private static class GenerateOutputsFunction extends RichMapFunction<Row, Row> {
-        private final Map<Param<?>, Object> paramMap;
-        private final String broadcastModelKey;
-        private final String[] inputCols;
-        private Pointer modelPointer = null;
-
-        private GenerateOutputsFunction(
-                Map<Param<?>, Object> paramMap, String broadcastModelKey, String[] inputCols) {
-            this.paramMap = new HashMap<>(paramMap);
-            this.inputCols = inputCols;
-            this.broadcastModelKey = broadcastModelKey;
+        @Override
+        public void open(Configuration parameters) throws Exception {
+            super.open(parameters);
+            modelPointer = ClinkJna.INSTANCE.OneHotEncoderModel_load(modelDataPath);
         }
 
         @Override
-        public Row map(Row row) throws IOException {
-            if (modelPointer == null) {
-                List<Tuple2<Integer, Integer>> modelDataList =
-                        getRuntimeContext().getBroadcastVariable(broadcastModelKey);
-                modelPointer = loadCppModel(paramMap, modelDataList);
-            }
-
+        public Row map(Row row) {
             Row resultRow = new Row(inputCols.length);
             for (int i = 0; i < inputCols.length; i++) {
                 String inputCol = inputCols[i];
```

Comment:
Would it be better to rename `int number` below to either `int value` or `int inputValue`?

Reply:
I think `inputValue` has the best readability among these choices. I'll make the change this way.